### PR TITLE
Appealsv2 wait time mg

### PR DESCRIPTION
**Pull request by markgreenburg**
- Adds new content to cards that show duration of the Next Event(s) in the appeal
- Adds logic to conditionally render these cards based on content
- Adds logic to display a top-level card above the NextEvents list when desired
- Adds ability to not show a duration card at all for a given NextEvent
- Refactors the DurationCards that show this info into their own component
- Adds tests for all the above

Top-Level duration card:
screen shot 2018-02-02 at 12 30 58 pm

Duration cards for each sub-event:
screen shot 2018-02-02 at 12 33 16 pm

When date range is exact, singular & plural months:
screen shot 2018-02-02 at 12 34 02 pm

### CODE REVIEW

File: src/js/claims-status/utils/appeals-v2-helpers.jsx [hunk 1]
```diff
@@ -565,6 +566,34 @@ const DECISION_REVIEW_CONTENT = (
...
 );
 
+/**
+ * Translates an array of two ints into a string that conveys a duration estimate
+ * @param {number[]} timeliness two integers that represent the low and high time durations
+ * (in months) of a given thing
+ * @returns {string} formatted to convey the estimated duration range, in months
+ */
+const makeDurationHeader = (timeliness) => {
+  if (!timeliness || !Array.isArray(timeliness) || timeliness.length !== 2) {
```
Comment (markgreenburg) [inline src/js/claims-status/utils/appeals-v2-helpers.jsx]: Not sure we need this, open to thoughts.

**Comment by cmgiven** [comment]
* "Months" should be lowercase in both the title and the description.
* Note the "Months months". Looking back at the issue, I realize this is my fault for leaving some stray "months" in the copy.
* In the description, the singular month should be formatted as "about 1 month" or "about 2 months". The ranges should be formatted as "between 2 and 4 months".

File: src/js/claims-status/utils/appeals-v2-helpers.jsx [hunk 2]
```diff
@@ -565,6 +566,34 @@ const DECISION_REVIEW_CONTENT = (
...
+    const durationError = new Error(
+      'vets_appeals_v2_helpers_makeDurationHeader_bad_timeliness_input'
+    );
+    Raven.captureException(durationError);
```
Comment (cvalarida) [inline src/js/claims-status/utils/appeals-v2-helpers.jsx]: Does it make more sense to just use `Raven.captureMessage()` here?

File: src/js/claims-status/utils/appeals-v2-helpers.jsx (hunk 1 above)
Comment (cvalarida) [reply src/js/claims-status/utils/appeals-v2-helpers.jsx]: Eh, I this it's good.

File: src/js/claims-status/utils/appeals-v2-helpers.jsx [hunk 3]
```diff
@@ -617,7 +655,10 @@ export function getNextEvents(currentStatus) {
           },
         ]
       };
-    case STATUS_TYPES.pendingForm9:
+    }
+    case STATUS_TYPES.pendingForm9: {
+      const certificationDuration = makeDurationHeader(details.certificationTimeliness);
+      const ssocDuration = makeDurationHeader(details.ssocTimeliness);
```
Comment (cvalarida) [inline src/js/claims-status/utils/appeals-v2-helpers.jsx]: Are these going to be camelCase like this, or snake_case? Might be worth putting a TODO here to make sure once we get integrated with the api.

**Review by cvalarida — APPROVED** [review]

**Comment by markgreenburg** [comment]
@cmgiven thanks for taking a look! Updated to:
- use lowercase 'month' / 'months'
- remove extra 'month' in description
- use 'about x month' / 'about x months' / 'between x–y months' in descriptions
- use en-dash instead of hyphens (😬 thanks for catching that)

Think this should be ready for a re-review

File: src/js/claims-status/utils/appeals-v2-helpers.jsx [hunk 4]
```diff
@@ -565,6 +566,46 @@ const DECISION_REVIEW_CONTENT = (
...
+
+  if (estIsExact && lowEst === 1) {
+    durationText.header = '1 month';
+    durationText.description = 'about 1 month';
+  } else if (estIsExact) {
+    durationText.header = `${lowEst} months`;
+    durationText.description = `about ${lowEst} months`;
+  } else {
+    durationText.header = `${lowEst}–${highEst} months`;
+    durationText.description = `between ${lowEst}–${highEst} months`;
```
Comment (cmgiven) [inline src/js/claims-status/utils/appeals-v2-helpers.jsx]: Thanks for the en-dash, but the description should use the word "and." :)

```
`between ${lowEst} and ${highEst} months`
```

**Comment by cmgiven** [comment]
One additional comment, then good to go!

File: src/js/claims-status/utils/appeals-v2-helpers.jsx (hunk 4 above)
Comment (markgreenburg) [reply src/js/claims-status/utils/appeals-v2-helpers.jsx]: Ah, right, thanks!

**Review by cvalarida — APPROVED** [review]
Changes look good.

File: src/js/claims-status/utils/appeals-v2-helpers.jsx (hunk 4 above)
Comment (markgreenburg) [reply src/js/claims-status/utils/appeals-v2-helpers.jsx]: Updated: https://github.com/department-of-veterans-affairs/vets-website/pull/7209/files#diff-07c91cd8e7a0729551ae3b4b7a393441R604

**Pull request merged**